src/functional/iterator/embedded.py

```diff
@@ -389,6 +389,25 @@ def make_in_iterator(inp, pos, offset_provider, *, column_axis):
 builtins.builtin_dispatch.push_key(EMBEDDED)  # makes embedded the default
 
 
+class ConstantField:
```

Comment:
Just an idea, but can you avoid introducing a new class and reuse `np_as_located_field(np.asarray(some_scalar))`? I just checked and at least construction seems to be fine. Otherwise I believe this should not be named constant field, maybe `ScalarField`?.

Reply:
`np_as_located_field(np.asarray(some_scalar))` is actually a function, which would try to create a field from an array with dimensions `np.array(some_scalar)`, which fails. Fundamentally `as_located_field` already gives the hint that it is a field with dimensions, which is defined in certain locations.

We can talk about the name but `ScalarField` is not a name I like:

 - A scalar field is a contradiction in terms (on the surface).
 - A field containing scalars, as opposed to multidimensional data could just as well be called scalar.
 - From the name it is not clear that the field is shiftable and accessible in every position and every dimension.

`ConstantField` describes a field that has the same value everywhere but could also be read differently. I think it is less confusing, but an even better name might be out there somewhere.

Reply:
I like constant field. Not that there is also `index_field`, you could implement it in the same way. (not saying that's better, but more consistent with what's already there.)